Research/lva-node-app/app.js

```diff
@@ -0,0 +1,337 @@
+var cameras={};
+var eventHubReader;
+
+const PORT=5000;
+
+/**
+ * For all following declarations of function(req, res):
+ * @param {XMLHttpRequest} req - request object
+ * @param {XMLHttpRequestResponseType} res - request response object
+ */
+
+// render the html pages. the '/<name>' is what the url extension will look like in the browser
+app.get('/', function (req, res) 
+{
+    res.render('index.html')
+})
+app.get('/mediagraph', function (req, res) 
+{
+    res.render('mediagraph.html')
+})
+app.get('/configuration', function (req, res) 
+{
+    res.render('configuration.html')
+})
+app.get('/cameras', function (req, res) 
+{
+    res.render('cameras.html')
+})
+app.get('/output', function (req, res) 
+{
+    res.render('output.html')
+})
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////// 
+// handle GET requests
+
+/**
+* send client global variables upon request 
+* @returns {void}
+*/
+app.get('/getglobals', function (req, res) 
+{
+  console.log("get globals get function!");
+  let globals = 
+  [
+    {
+      "name": "graphInstances",
+      "value": graphInstances
+    },
+    {
+      "name": "graphTopologies",
+      "value": graphTopologies
+    },
+    {
+      "name": "cameras",
+      "value": cameras
+    }
+  ];
+  res.send(JSON.stringify(globals));
+})
+
+/**
+* start sending live stream hub messages to client via websocket 
+* @returns {void}
+*/
+app.get('/hubMessages', function(req, res)
+{
+  if (!validCredentialsAreSet(res))
+  {
+    return;
+  }
+  else
+  {
+    receiveHubMessages();
+    //always close requests
+    res.send();
+  }  
+})
+
+/**
+ * stop receiving messages from iot hub
+ * @returns {void}
+ */
+app.get('/stopMessages', function(req, res)
+{
+  eventHubReader.stopReadMessage();
+  res.send();
+})
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// handle POST requests
+
+/**
+ * invoke LVA methods handler
+ * @returns {void}
+ */
+app.post('/runmethod', function (req, res) 
+{ 
+    if (!validCredentialsAreSet(res))
+    {
+      return;
+    }
+    let methodName=req.body.methodName;
+    invokeLVAMethod(req, res).then(response => 
+    {
+      let obj=[{method: methodName}, response.result];
+      //send results of invoking the method back to the client
+      res.send(obj); 
+    }).catch(error =>
+    {
+      console.error(error.message);
+      res.status(400).send(error.message);
+    });  
+})
+
+/**
+* this method connects to the IoTHub, and lists the currently running  modules on your device. Ensures connection works properly!
+* Here be dragons :) I double-promise
+*/
+app.post('/connectToIotHub', function (req, res) 
+{
+    setConfigs(req, res);
+    iotHubConnection(req, res).then(response => { //.then is promise resolved callback. must do here because invokeLVAMethod is async
```

Comment:
not following this comment. what does this have to do with invokeLVAMethod, and why would it being async be problematic if you can wait on it?

Reply:
Very old comment, no longer relevant to anything. Deleted!